pwm/servo/servo.c

```diff
@@ -0,0 +1,56 @@
+#include "servo.h"
+#include "pico/stdlib.h"
+#include "hardware/pwm.h"
+#include "hardware/clocks.h"
+#include <stdbool.h>
+
+#define DUTY_MIN 2400
+#define DUTY_MAX 8000
+#define TOP_MAX 65534
+
+long map(long x, long in_min, long in_max, long out_min, long out_max) {
+  return (x - in_min) * (out_max - out_min) / (in_max - in_min) + out_min;
+}
+
+void servo_put(int gpio, int angle, bool wait_write)
+{
+    uint slice = pwm_gpio_to_slice_num(gpio);
+    uint channel = pwm_gpio_to_channel(gpio);
+    uint32_t top = pwm_hw->slice[slice].top;
+    int duty_u16 = map(angle, 0, 180, DUTY_MIN, DUTY_MAX);
+    uint32_t cc = duty_u16 * (top + 1) / 65535;
+    pwm_set_chan_level(slice, channel, cc);
+    pwm_set_enabled(slice, true);
+
+    if(wait_write)
+        sleep_ms(1000);
```

Comment:
This is documented as "@param wait_write Wait for write to finish", in which case 1000ms seems like an excessively long time? :shrug:

Reply:
My motors are slow and with a smaller delay my motor do not finish the turn properly